framework/src/meshgenerators/MeshCollectionGenerator.C

```diff
@@ -0,0 +1,76 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "MeshCollectionGenerator.h"
+
+#include "CastUniquePointer.h"
+#include "MooseUtils.h"
+
+#include "libmesh/replicated_mesh.h"
+#include "libmesh/unstructured_mesh.h"
+
+registerMooseObject("MooseApp", MeshCollectionGenerator);
+
+template <>
+InputParameters
+validParams<MeshCollectionGenerator>()
+{
+  InputParameters params = validParams<MeshGenerator>();
+  params.addClassDescription("Collects multiple meshes into a single (unconnected) mesh.");
```

Comment:
OK - so you are very explicit about the fact that these are unconnected meshes - fine.